### ISSUE TITLE
Docs, settings, and types for extra loggers and appenders are a bit out of whack

### ISSUE THREAD

**Issue by steinybot** [sbt/sbt#7152]
The docs on how to [add a custom logger](https://www.scala-sbt.org/1.x/docs/Howto-Logging.html#Add+a+custom+logger) say to use `extraLoggers` but this is deprecated.

Presumably the replacement is `extraAppenders` however this is a `settingKey[AppenderSupplier]` where `AppenderSupplier` is:
```
trait AppenderSupplier {
  def apply(s: ScopedKey[_]): Seq[Appender]
}
```
where `Appender` is `sbt.internal.util.Appender` and not `XAppender` (`org.apache.logging.log4j.core.Appender`) as one would expect. This is internal so I'm hesitant to use it.

There is a `ConsoleAppenderFromLog4J` but this is also internal and `private[sbt]`.

It seems like `extraLoggers` was deprecated prematurely without a viable replacement.

**Comment by steinybot** [comment]
I tried using the deprecated `extraLoggers` and it doesn't seem to work. I took a look at the [PR that deprecated it](https://github.com/sbt/sbt/pull/5731) and it seems like maybe I would need to set `useLog4J := true`. This didn't seem to have any effect.

My ultimate goal is to modify the logging so that it includes the display name of the key for the task that is logging. I have the same task running in parallel on multiple sub-projects and it is really hard to tell which log message is for which project.

**Comment by clee704** [comment]
SBT log messages only have a log level and a message, making it difficult to work with multi-module project. At least it should include the module name. Task name / timestamp would be nice to have.

**Comment by clee704** [comment]
If you don't care about using the internal API, you can use the following code to print timestamp and the task being run:

```
logManager := {
  import sbt.Def.ScopedKey
  import sbt.internal.LogManager
  import sbt.internal.util.MainAppender._
  import java.time.ZonedDateTime
  import java.time.format.DateTimeFormatter

  class FormattedWriter(formatter: String => String)
      extends java.io.PrintWriter(System.out) {
    override def print(msg: String): Unit = System.out.print(formatter(msg))
    override def println(msg: String): Unit = System.out.println(formatter(msg))
  }

  val datetimeFormatter = DateTimeFormatter.ofPattern("uuuu-MM-dd HH:mm:ss.SSSXXX")
  val withTimestamp = (msg: String) => {
    val timestamp = ZonedDateTime.now().format(datetimeFormatter)
    s"$timestamp $msg"
  }

  LogManager.withLoggers(
    (key: ScopedKey[_], state: State) => {
      def unwrapStreamsKey(key: ScopedKey[_]): ScopedKey[_] = key.scope.task match {
        case Select(task) => ScopedKey(key.scope.copy(task = Zero), task)
        case _            => key // should never get here
      }
      val taskName = Project.showContextKey(state).show(unwrapStreamsKey(key)).replaceAll(" / ", "/")
      val withTask = (msg: String) => s"$taskName $msg"
      val console = ConsoleOut.printWriterOut(new FormattedWriter(s => withTimestamp(withTask(s))))
      defaultScreen(console, LogManager.suppressedMessage(key, state))
    },
    extra = extraAppenders.value
  )
}
```
